components/tikv_util/src/mpsc/batch.rs

```diff
@@ -118,6 +118,7 @@ impl<T> Clone for Sender<T> {
 impl<T> Drop for Sender<T> {
     #[inline]
     fn drop(&mut self) {
+        drop(self.sender.take());
```

Comment:
can we use without option and just drop(self.sender) directly?

Reply:
No, `drop` can't move `self`, so an Option is necessary.

Reply:
Use [ManuallyDrop](https://doc.rust-lang.org/beta/std/mem/struct.ManuallyDrop.html) should be better since it is zero overhead.